generator/booking_quality_check/booking_addr_match.cpp

```diff
@@ -0,0 +1,135 @@
+#include "generator/booking_dataset.hpp"
+#include "generator/osm_source.hpp"
+
+#include "map/framework.hpp"
+
+#include "indexer/classificator_loader.hpp"
+#include "indexer/data_header.hpp"
+#include "indexer/index.hpp"
+#include "indexer/mwm_set.hpp"
+
+#include "platform/country_file.hpp"
+#include "platform/local_country_file.hpp"
+#include "platform/local_country_file_utils.hpp"
+#include "platform/platform.hpp"
+
+#include "geometry/distance_on_sphere.hpp"
+
+#include "search/processor_factory.hpp"
+#include "search/ranking_info.hpp"
+#include "search/result.hpp"
+#include "search/search_quality/helpers.hpp"
+#include "search/search_tests_support/test_search_engine.hpp"
+#include "search/search_tests_support/test_search_request.hpp"
+
+#include "storage/country_info_getter.hpp"
+#include "storage/index.hpp"
+#include "storage/storage.hpp"
+
+#include "coding/reader_wrapper.hpp"
+
+#include "std/fstream.hpp"
+#include "std/iostream.hpp"
+#include "std/numeric.hpp"
+#include "std/shared_ptr.hpp"
+
+#include "3party/gflags/src/gflags/gflags.h"
+
+using namespace generator;
+using namespace storage;
+using namespace search;
+using namespace search::tests_support;
+
+void DidDownload(TCountryId const & /* countryId */,
+                 shared_ptr<platform::LocalCountryFile> const & /* localFile */)
+{
+}
+
+bool WillDelete(TCountryId const & /* countryId */,
+                shared_ptr<platform::LocalCountryFile> const & /* localFile */)
+{
+  return false;
+}
+
+uint64_t ReadVersionFromHeader(platform::LocalCountryFile const & mwm)
+{
+  vector<string> specialFiles = {
+    WORLD_FILE_NAME,
+    WORLD_COASTS_FILE_NAME,
+    WORLD_COASTS_OBSOLETE_FILE_NAME
+  };
+  for (auto const & name : specialFiles)
+  {
+    if (mwm.GetCountryName() == name)
+      return mwm.GetVersion();
+  }
+
+  ModelReaderPtr reader = FilesContainerR(mwm.GetPath(MapOptions::Map)).GetReader(VERSION_FILE_TAG);
```

Comment:
Why not to use DataHeader to read version? Of course, you need to slightly modify DataHeader but the code here will be cleaner.